src/main/java/co/novu/common/rest/IdempotencyKeyInterceptor.java

```diff
@@ -0,0 +1,29 @@
+package co.novu.common.rest;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class IdempotencyKeyInterceptor implements Interceptor{
```

Comment:
We don't necessarily need a dedicated Interceptor for this. We can make it a dynamic header by using @ Headers for the POST and PATCH calls

Reply:
If we don't use interceptor for adding Idempotency header, do I need to add @Header annotation for POST and PATCH calls in all *Api classes? as that works on method level